### PR TITLE
Added user layout for ergodox

### PR DESCRIPTION
**Pull request by pvazteixeira**
## Types of Changes

- [ ] Core
- [ ] Bugfix
- [ ] New feature
- [ ] Enhancement/optimization
- [ ] Keyboard (addition or update)
- [x] Keymap/layout/userspace (addition or update)
- [ ] Documentation

## Checklist

- [x] My code follows the code style of this project: [**C**](https://docs.qmk.fm/#/coding_conventions_c), [**Python**](https://docs.qmk.fm/#/coding_conventions_python)
- [x] I have read the [**PR Checklist** document](https://docs.qmk.fm/#/pr_checklist) and have made the appropriate changes.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [ ] I have added tests to cover my changes.
- [x] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

File: keyboards/ergodox_ez/keymaps/pvazteixeira/keymap.c [hunk 1]
```diff
@@ -0,0 +1,218 @@
+//pvazteixeira's ergodox (classic) layout
```
Comment (drashna) [inline keyboards/ergodox_ez/keymaps/pvazteixeira/keymap.c:1]: Could you add a GPL2+ compatible license header here? 

For instance: 
https://github.com/qmk/qmk_firmware/blob/28ff51175b6d45ce5a8dbfe7ed6e7a12df3bc8a8/quantum/template/base/keyboard.c#L1-L15

File: keyboards/ergodox_ez/keymaps/pvazteixeira/keymap.c [hunk 2]
```diff
@@ -0,0 +1,218 @@
...
+
+#include QMK_KEYBOARD_H
+#include "version.h"
+
+#define BASE 0 // default layer
+#define SYMB 1 // symbols
+#define MDIA 2 // media keys
```
Comment (drashna) [inline keyboards/ergodox_ez/keymaps/pvazteixeira/keymap.c:8]:
```suggestion
enum keyboard_layers (
    BASE,
    SYMB,
    MDIA,
};
```

File: keyboards/ergodox_ez/keymaps/pvazteixeira/keymap.c [hunk 3]
```diff
@@ -0,0 +1,218 @@
...
+  KC_TRNS, KC_TRNS, KC_WBAK
+),
+};
+
+bool process_record_user(uint16_t keycode, keyrecord_t *record) {
+  if (record->event.pressed) {
+
+    // Log event
+    uprintf ("KL: col=%02d, row=%02d, pressed=%d, layer=%d\n", record->event.key.col,
+             record->event.key.row, record->event.pressed, biton32(layer_state));
```
Comment (drashna) [inline keyboards/ergodox_ez/keymaps/pvazteixeira/keymap.c:152]:
```suggestion
             record->event.key.row, record->event.pressed, get_highest_layer(layer_state));
```

File: keyboards/ergodox_ez/keymaps/pvazteixeira/keymap.c [hunk 4]
```diff
@@ -0,0 +1,218 @@
...
+};
+
+// Runs whenever there is a layer state change.
+layer_state_t layer_state_set_user(layer_state_t state) {
+  ergodox_board_led_off();
+  ergodox_right_led_1_off();
+  ergodox_right_led_2_off();
+  ergodox_right_led_3_off();
+
+  uint8_t layer = biton32(state);
```
Comment (drashna) [inline keyboards/ergodox_ez/keymaps/pvazteixeira/keymap.c:183]:
```suggestion
  uint8_t layer = get_highest_layer(state);
```

File: keyboards/ergodox_ez/keymaps/pvazteixeira/readme.md [hunk 5]
```diff
@@ -0,0 +1,12 @@
+# pvazteixeira's ergodox layout 
+  
+This layout uses Mod Tap to overload the thumb clusters and make it easier to use with both spacemacs and regolith desktop.
+The outermost columns are mostly unused to reduce little finger stress.
+
+## Uploading the firmware
+
+```sh
+teensy_loader_cli --mcu=TEENSY2 -w -v ergodox_ez_pvazteixeira.hex
+```
```
Comment (drashna) [inline keyboards/ergodox_ez/keymaps/pvazteixeira/readme.md:10]: This isn't needed anymore. 

```suggestion
```

**Review by drashna — COMMENTED** [review]
I didn't mark the infinity, but you could actually move these to `layouts/community/ergodox/pvazteixeira/`, and it will be used by both boards, actually. 

https://docs.qmk.fm/#/feature_layouts

**Comment by Erovia** [comment]
I urge you to take drashna's advice and share your keymap between the boards.

You can handle differences with `#if defined(KEYBOARD_ergodox_ez)` and `#if defined(KEYBOARD_ergodox_infinity)`.

**Comment by pvazteixeira** [comment]
Hey @drashna (and @Erovia) - thanks for the feedback and suggestions, I'll push the changes in the coming week!

**Review by drashna — APPROVED** [review]

**Comment by pvazteixeira** [comment]
@drashna @Erovia thanks again for the feedback! Do you have any pointers on how to work with the LCD and LCD backlight on the infinity? The docs only seem to cover other displays

File: layouts/community/ergodox/pvazteixeira/README.md [hunk 6]
```diff
@@ -0,0 +1,14 @@
+
+# pedro's ergodoxes
```
Comment (fauxpark) [inline layouts/community/ergodox/pvazteixeira/README.md:2]:
```suggestion
# pedro's ergodoxes
```
Please also rename this file to lowercase.

**Review by fauxpark — CHANGES_REQUESTED** [review]

File: layouts/community/ergodox/pvazteixeira/README.md (hunk 6 above)
Comment (pvazteixeira) [reply layouts/community/ergodox/pvazteixeira/README.md]: fixed in 225a260, 8cd7686. thanks

File: layouts/community/ergodox/pvazteixeira/keymap.c [hunk 7]
```diff
@@ -0,0 +1,168 @@
...
+    /*             if (prev_is_alpha == false && is_lower) { */
+    /*                 *w -= ('a' - 'A'); */
+    /*             } */
+    /*             prev_is_alpha = is_lower || is_upper; */
+    /*             w++; */
+    /*         } */
+    /*     } */
+    /* } */
+};
+
```
Comment (fauxpark) [inline layouts/community/ergodox/pvazteixeira/keymap.c:145]:
```suggestion
```

These can be removed if you're not using them.

**Review by fauxpark — CHANGES_REQUESTED** [review]

**Pull request closed without merging**